### DISCUSSION TITLE
Generate {} instead of object in OpenAPI spec

### DISCUSSION THREAD

**Discussion by Hes-Siemelink** [smallrye/smallrye-open-api#1361] [Q&A] [answered]
Hello all

I am generating the OpenAPI spec from java.ws.rs with the gradle plugin.

I have a container object that has an accessor method `getValue()` that returns Object. This value can be an Integer, Boolean, List, etc. The OpenAPI spec generates `object` for this. But when generating the client bindings in Python, it would be more useful to have this as `{}`, so the response JSON can be parsed correctly in the case of Integer or Boolean.

Is there a way to generate the correct return type? For example by adding an annotation to the Java method.

Thanks!

**Comment by MikeEdgar** [comment, marked as answer]
Hi @Hes-Siemelink , this should work as you expect when https://github.com/smallrye/smallrye-open-api/pull/1347 is merged and released. I hope/expect it should be in the next version.

**Reply by MikeEdgar** [reply]
Also, there is a work-around in https://github.com/smallrye/smallrye-open-api/issues/1022 if you want to get going immediately.

**Reply by Hes-Siemelink** [reply]
Thanks!

One thing, I am not sure where to configure this property when using the gradle plugin:
```
mp.openapi.schema.java.lang.Object={ "name": "AnyType" }
```
Should that be in `build.gradle` or somewhere else?

Thanks again

**Reply by MikeEdgar** [reply]
@Hes-Siemelink are you using an `application.properties` file? You should be able to configure the path to that properties using the `configProperties` plugin configuration property.

**Reply by Hes-Siemelink** [reply]
Thanks, missed that one! 

Got my gradle script to work and it doesn't complain about not being able to find `application.properties` anymore.

Next hurdle is how to configure it correctly... I suppose that I need to put something specific into the `"name"` part of the following snippet, but what exactly? 

```
mp.openapi.schema.java.lang.Object={ "name": "AnyType" }
```

I have the following schema being generated:

```
    Variable:
      type: object
      properties:
        title:
          type: string
        key:
          type: string
        value:
          type: object
```

And I want `value` to be 

```
        value:
          type: {}
```  

It doesn't seem to work with the above `application.properties`. My apologies, I am not very familiar with the tooling so guessing half the time.

Thanks

**Reply by MikeEdgar** [reply]
The `name` is just whatever key you want to be used for the `{}` entry in `/components/schemas` which will be used in a `$ref`. How is `value` defined in your code? In this case if it's a `java.lang.Object` I would expect to see the output below. The `type` of `value` won't be `{}` itself, but rather simply omitted via the reference.
```yaml
value:
  $ref: '#/components/schemas/AnyType'
```

**Reply by Hes-Siemelink** [reply]
My `ReleaseApi` class defines the variables endpoint:

```java

    @GET
    @Path("/{releaseId:.*Release[^/]*}/variables")
    @PublicApiMember
    List<Variable> getVariables(@PathParam("releaseId") String releaseId);
```

`Variable` is defined as follows:

```java
public class Variable {

    private Object value;

    public Object getValue() {
        return value;
    }

   // more stuff
}
```

The OpenAPI bit is 

```yaml
  /api/v1/releases/{releaseId}/variables:
    get:
      operationId: ReleaseApi_getVariables
      parameters:
      - name: releaseId
        in: path
        required: true
        schema:
          pattern: ".*Release[^/]*"
          type: string
      responses:
        "200":
          description: OK
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/Variable'
```

And the schema definition of Variable is:

```yaml
    Variable:
      type: object
      properties:
        value:
          type: object        # <== This should be {}

      // More stuff
```

**Reply by MikeEdgar** [reply]
I see what you mean. I'll take a look, but maybe there's a bug that is preventing `value` from being a `$ref`. Do you see a schema for `AnyType` ?

**Reply by Hes-Siemelink** [reply]
No schema entry for `AnyType`.

And no endpoints return `Object` directly, it is all through return types.

**Reply by MikeEdgar** [reply]
Darn... `configProperties` doesn't seem to be loaded for the Gradle plugin. We'll need a fix for that.

**Reply by Hes-Siemelink** [reply]
Oh dear. It does complain (blow up) when I specify a file that doesn't exist.

**Comment by Hes-Siemelink** [comment]
Cool, thanks!
